src/plugin.ts

```diff
@@ -48,7 +48,24 @@ export class Auth0Plugin implements Auth0VueClient {
   constructor(
     private clientOptions: Auth0VueClientOptions,
     private pluginOptions?: Auth0PluginOptions
-  ) {}
+  ) {
+    const plugin = this;
+
+    // Vue Plugins can have issues when passing around the instance to `provide`
+    // Therefor we need to bind all methods correctly to `this`.
+    this.loginWithRedirect = this.loginWithRedirect.bind(plugin);
+    this.loginWithPopup = this.loginWithPopup.bind(plugin);
+    this.logout = this.logout.bind(plugin);
+    this.getAccessTokenSilently = this.getAccessTokenSilently.bind(plugin);
+    this.getAccessTokenWithPopup = this.getAccessTokenWithPopup.bind(plugin);
+    this.checkSession = this.checkSession.bind(plugin);
+    this.handleRedirectCallback = this.handleRedirectCallback.bind(plugin);
+    this.buildAuthorizeUrl = this.buildAuthorizeUrl.bind(plugin);
+    this.buildLogoutUrl = this.buildLogoutUrl.bind(plugin);
+    this.__checkSession = this.__checkSession.bind(plugin);
+    this.__refreshState = this.__refreshState.bind(plugin);
+    this.__proxy = this.__proxy.bind(plugin);
```

Comment:
Just a question, is there a dynamic way we can do this (e.g. find all functions that are direct members of `this` and bind it)? I can just see this being missed later; if we add new methods later we're likely to forget to add it to this list and run into issues when someone tries to use it.

Reply:
Updated the PR to do it more dynamic.

Reply:
Very cool